crates/sui-indexer/src/handlers/mod.rs

```diff
@@ -162,6 +164,16 @@ impl<T> CommonHandler<T> {
                 self.handler.set_watermark_hi(committer_watermark).await?;
                 tuple_batch = vec![];
             }
+
+            if let Some(end_checkpoint) = end_checkpoint_opt {
+                if next_cp_to_process > end_checkpoint {
+                    tracing::info!(
+                        "Reached end checkpoint, stopping handler {}...",
+                        self.handler.name()
+                    );
+                    return Ok(());
```

Comment:
anticipated q: similarly, break here does not stop the binary entirely, the ingestion framework part is still running for snapshot, which is fine for benchmark.